### PR TITLE
Improved completion provider's handling of method overrides so it pro…

### PR DESCRIPTION
**Pull request by erictraut**
…perly handles properties.

### CODE REVIEW

File: packages/pyright-internal/src/languageService/completionProvider.ts [hunk 1]
```diff
@@ -926,15 +959,21 @@ export class CompletionProvider {
...
-        const parameters = getParameters(declaredType, decl);
+        const parameters = getParameters();
         if (decl.node.name.value !== '__init__') {
             sb += 'return ';
         }
 
+        if (isProperty) {
+            // If it's a property, it's not clear what to provide in the
+            // implementation. Assume that the user will fill in the expression.
+            return sb;
```
Comment (jakebailey) [inline packages/pyright-internal/src/languageService/completionProvider.ts:970]: It'd actually be `return super().prop`, if we did want to generate something.

Comment (jakebailey) [reply packages/pyright-internal/src/languageService/completionProvider.ts:970]: Or did you see an ambiguity in this case?

Comment (erictraut) [reply packages/pyright-internal/src/languageService/completionProvider.ts]: Yeah, that's reasonable.

**Comment by jakebailey** [comment]
This seems good as is for getters. I suppose we don't have anything for setters or anything, given to do overrides of that you have to do:

```python
class A:
    @property
    def foo(self): return ...

    @foo.setter
    def foo(self, x): ...


class B(A):
    @A.foo.setter
    def foo(self, x): ...
```

So hopefully we offer the completion `A.foo.setter` alright.

**Review by jakebailey — APPROVED** [review]

**Pull request merged**